projects/plugins/jetpack/modules/masterbar/admin-menu/class-base-admin-menu.php

```diff
@@ -221,6 +253,54 @@ public function enqueue_scripts() {
 		);
 	}
 
+	/**
+	 * Hide the submenu page based on slug and return the item that was hidden.
+	 *
+	 * Instead of actually removing the submenu item, a safer approach is to hide it and filter it in the API response.
+	 * In this manner we'll avoid breaking third-party plugins depending on items that no longer exist.
+	 *
+	 * A false|array value is returned to be consistent with remove_submenu_page() function
+	 *
+	 * @param string $menu_slug The parent menu slug.
+	 * @param string $submenu_slug The submenu slug that should be hidden.
+	 * @return false|array
+	 */
+	public function hide_submenu_page( $menu_slug, $submenu_slug ) {
+		global $submenu;
+
+		if ( ! isset( $submenu[ $menu_slug ] ) ) {
+			return false;
+		}
+
+		foreach ( $submenu[ $menu_slug ] as $i => $item ) {
+			if ( $submenu_slug === $item[2] ) {
+				$css_classes = isset( $item[4] ) ? $item[4] . '  hide-if-js hide-if-no-js' : 'hide-if-js hide-if-no-js';
+				// phpcs:ignore WordPress.WP.GlobalVariablesOverride.Prohibited
+				$submenu[ $menu_slug ][ $i ][4] = $css_classes;
+				return $item;
+			}
+		}
+
+		return false;
+	}
+
+	/**
+	 * Check if the menu has submenu items visible
+	 *
+	 * @param array $submenu_items The submenu items.
+	 * @return bool
+	 */
+	public function has_visible_items( $submenu_items ) {
+		$visible_items = array_filter(
+			$submenu_items,
+			static function ( $item ) {
+				return empty( $item[4] ) || strpos( $item[4], 'hide-if-js hide-if-no-js' ) === false;
```

Comment:
Can we move the `hide-if-js hide-if-no-js` in a variable since it's reused plenty of times and we should make sure it matches here?

Reply:
Yup, good catch, thanks! I've moved it in a const.